### DISCUSSION TITLE
Boot logo executable???

### DISCUSSION THREAD

**Discussion by wiebenor** [Ralim/IronOS#1133] [Q&A] [answered]
Please forgive me if I'm missing something, but I  know there used to be (and still is, I believe)an executable for windows to convert .BMP files into a bootlogo firmware file, but when I search on Google I mainly find the command line creation tool instructions instead,(or at least, very easily) and I'm wondering why, when you do find where it's at, you have to go back to very old codebases to find the executable? Is there something inherently bad about that method, to where I almost have to pull teeth to find the windows version, or a link to it? I understand that most are comfortable enough to either use the WSL or Linux command line, but from what I understand, the windows executable works, or at least, used to, and I for one prefer the 1-click method that had, although I haven't actually tried adding a bootlogo to my iron yet, because I'm not 100% sure if I want to...

**Comment by Ralim** [comment, marked as answer]
The windows executable does _not_ work with versions of IronOS from the last year or so.
It was dropped as it did a sub-par job of the conversion, and required windows to build/use (which I no longer have installed).
Decision was made to just support the one tool as python is cross platform and is far easier to maintain

**Reply by wiebenor** [reply]
Ah, thanks... Any idea if the instructions on windows, using python or whatever would be the same, or similar??? I might be able to build one that way if so. Also, I have only glanced at the python based instructions since I wasn't really wanting to try and comprehend all that, at that time, and was wondering if it's a script, or if not, if it could be made into one? I'm not very well versed in the Linux based command line as I rarely use Linux, or the associated command line, so I'm not sure exactly how all that works

**Reply by Ralim** [reply]
Should be practically identical. It works as as script with just a few command line args.
I assume one Windows it would work nearly identically, though maybe you need to run it as `python3.exe <script name and args>`.
I dont have a windows machine around to test on but if you give it a try and have issues feel free to comment back the errors and I'll give you a hand

**Comment by River-Mochi** [comment]
@wiebenor  if you want, I have python installed on Windows and I could kind of show you how I did it so you can convert some boot logos.  or you could give me the images and I could convert them for you?    I'm new to python but I figured out how to do it with the python script in this github. the instructions in the  Boot Logo documentation did not work for me so  I did  a bunch of google search on how to get python and pillow to work /install in windows and got it to work. 
I have converted severage logos to dfu now and successfully installed them. 

Start with the document on boot logo in this github though and see if you can get python 3 and pillow installed.  I found that the Powershell worked better /easier for me in windows than using python from Command prompt.
